packages/drivers/odsp-driver/src/retryUtils.ts

```diff
@@ -0,0 +1,60 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ITelemetryLogger } from "@fluidframework/common-definitions";
+import { assert, delay } from "@fluidframework/common-utils";
+import { canRetryOnError } from "@fluidframework/driver-utils";
+import { OdspErrorType } from "@fluidframework/odsp-driver-definitions";
+import { Odsp409Error } from "./epochTracker";
+
+/**
+ * This method retries only for retriable coherency and service read only errors.
+ */
+export async function runWithRetry<T>(
+    api: () => Promise<T>,
+    callName: string,
+    logger: ITelemetryLogger,
+    checkRetry?: () => void,
+): Promise<T> {
+    let retryAfter = 1000;
+    const start = performance.now();
+    for (let retry = 1; ; retry++) {
+        try
+        {
+            return await api();
+        } catch (error) {
+            if (checkRetry !== undefined) {
```

Comment:
Should this check be before actual call to api()? Maybe in both places?

Reply:
I think just before is fine.